services/user.service.ts

```diff
@@ -135,29 +143,35 @@ export async function inquiryService(userId) {
 
 
 // 내 정보 수정
-export async function editUserService(userId, data = {} ) {
-  const { password, nickname, email, image} = data;
+export async function editUserService(userId: number, data = {} ): Promise<{
+  id: number;
+  email: string;
+  nickname: string;
+  image: string | null;
+  createdAt: Date;
+  updatedAt: Date;
+}> {
+  const { password, nickname, email, image} = data as {
```

Comment:
as로 타입단언을 하면 기존 타입을 무시하게 되어서 인자에 있는 data타입을 따로 정의해서 쓰는게 더 좋을 것 같아요~

Reply:
data 타입을 따로 정의하면 맞지 않는 부분들이 있어서 서비스,컨트롤러 모두 다음과 같이 수정했습니다.
컨트롤러를 수정하려고 보니 여기도 as 가 사용되어 있는데 이것도 변경해야하는건지? 궁금합니다.

```suggestion
  // 내 정보 수정
  type EditUserData = {
    password: string;
    nickname: string;
    email: string;
    image: string | null ;
};

export async function editUserService(userId: number, data: EditUserData ): Promise<{
  id: number;
  email: string;
  nickname: string;
  image: string | null;
  createdAt: Date;
  updatedAt: Date;
}> {
  const { password, nickname, email, image } = data;
  

 - 컨트롤러 
    // 수정할 값들 꺼내오기
    const { password, nickname, email, image } = req.body as {
      password: string;
      nickname: string;
      email: string;
      image: string | null;
    };
```